topostats/tracing/dnatracing.py

```diff
@@ -423,34 +420,32 @@ def get_splined_traces(self):
             # LOGGER.info(f"nbr             : {nbr}")
             # LOGGER.info(f"step_size       : {step_size}")
             # LOGGER.info(f"self.pixel_size : {self.pixel_size}")
-            while nbr / step_size < 4:
-                if step_size <= 1:
-                    step_size = 1
+            while nbr / step_size_px < 4:
+                if step_size_px <= 1:
+                    step_size_px = 1
                     break
-                step_size = -1
-            if self.mol_is_circular[dna_num]:
+                step_size_px = -1
 
-                # if nbr/step_size > 4: #the degree of spline fit is 3 so there cannot be less than 3 points in splined trace
+            if self.mol_is_circular[dna_num]:
 
-                # ev_array = np.linspace(0, 1, nbr * step_size)
-                ev_array = np.linspace(0, 1, int(nbr * step_size))
+                ev_array = np.linspace(0, 1, nbr * step_size_px)
 
-                for i in range(step_size):
+                for i in range(step_size_px):
```

Comment:
The advantage of having a method that works with a single molecule is that you then don't need to have all the complicated logic here to extract the arrays.

Reply:
If I recall correctly, this is for sampling the coordinate array so that the splining can be made smoother, and this mechanism should still be in place even with a method for a single molecule